crates/swc_css_parser/src/lexer/mod.rs

```diff
@@ -542,7 +542,29 @@ where
                 }
                 // Otherwise, consume a url token, and return it.
                 _ => {
-                    return self.read_url();
+                    return self.read_url().map(|mut token| {
```

Comment:
Why does `read_url` return an incomplete token?
I think accepting `name` as a parameter and returning something in a valid state is better

Reply:
No partial-init is one of philosophies of rust.
Is there a drawback of this approach?

Reply:
Yep, initially I do it, but name in one function, but token creation in other, so I tried to keep them independent, but it is not problem and I can refactor it

Reply:
Can you do the refactor? I think it's a better way to go.

Reply:
Yes, I will do it in near future :+1:

Reply:
Thank you! I think we should follow the philosophy of the language if we can.

Reply:
@kdy1 Done :+1: Found couple TODO with `url`, I will fix them in the next PR